torch/lib/c10d/ProcessGroupNCCL.cpp

```diff
@@ -194,11 +266,81 @@ ProcessGroupNCCL::ProcessGroupNCCL(
   processGroupID_ = std::to_string(processGroupCounterMap_[groupKey]);
   groupPgID_ = groupName_ + "_" + processGroupID_;
   pgUniqueNCCLIDCnt_[groupPgID_] = -1;
+  ncclCommWatchdogThread_ =
```

Comment:
I am debating with myself on whether this should be optional and whether we should allow users to control this. If an application is never bothered by NCCL comm errors, will this still be useful? If not, does it make sense to provide an arg to avoid always having an additional thread?